File: Koflit/Quicksort.js
// naive solution
const quickSort = function (arr) {
  if (arr.length <= 1) return arr;

  const pivot = arr[0];
  const left = [];
  const right = [];

  for (let i = 1; i < arr.length; i++) {
    if (arr[i] <= pivot) left.push(arr[i]);
    else right.push(arr[i]);
  }

  const lSorted = quickSort(left);
  const rSorted = quickSort(right);
  return [...lSorted, pivot, ...rSorted];
};

const quick = quickSort([1, 2, 43, 100, 21]);
console.log(quick);

File: Algo/python/QuickSort.py
def QuickSort(array , start , end):
  if start >= end:
    return
  pivot = start
  left = start + 1
  right = end

  while (left <= right):
    while(left <= end and array[left] <= array[pivot]):
      left += 1
    while(right > start and array[right] >= array[pivot]):
      right -= 1
    if (left > right):
      array[left], array[right] = array[right] , array[left] 
    else:
      array[left] , array[right] = array[right], array[left]
  QuickSort(array, start, right-1)
  QuickSort(array, right+1, end)

array = [3,2,5,5,1,2,3,5,9,10,2,25]
QuickSort(array, 0, len(array)-1)
print(array)